pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,75 @@
+package dns
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		zones  []configv1.DNSZone
+		expect []operatorv1.ConditionStatus
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: []operatorv1.ConditionStatus{operatorv1.ConditionFalse},
+		},
+		{
+			name:   "when no zones available",
+			zones:  []configv1.DNSZone{},
+			expect: []operatorv1.ConditionStatus{""},
```

Comment:
I think that should be `[]operatorv1.ConditionStatus{""}` (or `[]string{}` if you change the type).